addons/udes_stock/models/stock_picking.py

```diff
@@ -1342,7 +1345,24 @@ def _new_picking_for_group(self, group_key, moves, **kwargs):
             # NB: only workable pickings
             ('state', 'in', ['assigned', 'confirmed', 'waiting']),
         ])
+
+        if not picking and picking_for_reuse:
+            # Otherwise use the suggested picking
+            if (picking_for_reuse.picking_type_id.id != picking_type.id or
+                picking_for_reuse.location_id.id != src_loc.id or
+                picking_for_reuse.location_dest_id.id != dest_loc.id or
+                picking_for_reuse.state not in
+                    ('assigned', 'confirmed', 'waiting')):
+                raise ValueError(_('Cannot reuse picking because it is not '
```

Comment:
This changes the current behaviour, if we cannot find a picking we create a new one. Therefore if the picking to reuse does not meet the search domain we should create a new picking.